code/datums/status_effects/debuffs.dm

```diff
@@ -69,17 +69,21 @@
 /datum/status_effect/incapacitating/sleeping/tick()
 	if(owner.maxHealth)
 		var/health_ratio = owner.health / owner.maxHealth
-		if(health_ratio > 0.8)
-			var/healing = -0.2
-			if((locate(/obj/structure/bed) in owner.loc))
-				healing -= 0.3
-			else
-				if((locate(/obj/structure/table) in owner.loc))
-					healing -= 0.1
+		var/healing = -0.2
+		if((locate(/obj/structure/bed) in owner.loc))
+			healing -= 0.3
+		else if((locate(/obj/structure/table) in owner.loc))
+			healing -= 0.1
+		for(var/obj/item/bedsheet/bedsheet in range(owner.loc,0))
+			if(bedsheet.loc != owner.loc) //bedsheets in your backpack/neck don't give you comfort
```

Comment:
if both of them are on the same turf they'll have the same `loc`

Reply:
I think thats what he's doing here. He's checking if they're not on the same turf, (Aka its in his bag), and then doesnt use that one.

Reply:
u rite